### ISSUE TITLE
--no-namespace option ignored

### ISSUE THREAD

**Issue by epavlic-revenue** [forcedotcom/cli#2855]
### Summary

- `sf` 2.30.8
- `project-scratch-def` with namespace provided
- `sf org create scratch --definition-file config/project-scratch-def.json --no-namespace` creates a scratch org with namespace

### Expected result
Created scratch org should not have a namespace

### Actual result
Created scratch org does have a namespace

### System Information

```json
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.30.8",
  "nodeVersion": "node-v20.5.1",
  "osVersion": "Darwin 23.4.0",
  "rootPath": "/opt/homebrew/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.9 (core)",
    "@oclif/plugin-commands 3.1.4 (core)",
    "@oclif/plugin-help 6.0.13 (core)",
    "@oclif/plugin-not-found 3.0.11 (core)",
    "@oclif/plugin-plugins 4.2.5 (core)",
    "@oclif/plugin-search 1.0.16 (core)",
    "@oclif/plugin-update 4.1.13 (core)",
    "@oclif/plugin-version 2.0.12 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.12 (core)",
    "@oclif/plugin-which 3.1.0 (core)",
    "@salesforce/cli 2.30.8 (core)",
    "apex 3.0.25 (core)",
    "auth 3.3.13 (core)",
    "data 3.1.1 (core)",
    "deploy-retrieve 3.2.16 (core)",
    "devops-center 1.2.0 (user)",
    "info 3.0.27 (core)",
    "limits 3.1.10 (core)",
    "marketplace 1.0.25 (core)",
    "org 3.3.14 (core)",
    "packaging 2.1.0 (user)",
    "schema 3.1.3 (core)",
    "settings 2.0.27 (core)",
    "sobject 1.1.13 (core)",
    "source 3.1.15 (core)",
    "telemetry 3.1.13 (core)",
    "templates 56.0.16 (core)",
    "trust 3.3.10 (core)",
    "user 3.2.11 (core)",
    "sfdx-ext 0.1.4 (user)"
  ]
}
```

**Comment by epavlic-revenue** [comment]
I'm still seeing the issue in the latest release:

```json
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.39.6",
  "nodeVersion": "node-v20.5.1",
  "osVersion": "Darwin 23.4.0",
  "rootPath": "/opt/homebrew/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.16 (core)",
    "@oclif/plugin-commands 3.3.1 (core)",
    "@oclif/plugin-help 6.0.21 (core)",
    "@oclif/plugin-not-found 3.1.6 (core)",
    "@oclif/plugin-plugins 5.0.14 (core)",
    "@oclif/plugin-search 1.0.23 (core)",
    "@oclif/plugin-update 4.2.7 (core)",
    "@oclif/plugin-version 2.0.17 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.15 (core)",
    "@oclif/plugin-which 3.1.8 (core)",
    "@salesforce/cli 2.39.6 (core)",
    "apex 3.1.9 (core)",
    "auth 3.6.3 (core)",
    "data 3.3.2 (core)",
    "deploy-retrieve 3.6.6 (core)",
    "devops-center 1.2.0 (user)",
    "info 3.2.3 (core)",
    "limits 3.3.4 (core)",
    "marketplace 1.2.4 (core)",
    "org 4.1.3 (core)",
    "packaging 2.1.0 (user)",
    "schema 3.3.4 (core)",
    "settings 2.2.1 (core)",
    "sobject 1.3.6 (core)",
    "source 3.3.3 (core)",
    "telemetry 3.3.4 (core)",
    "templates 56.2.4 (core)",
    "trust 3.6.6 (core)",
    "user 3.5.2 (core)",
    "sfdx-ext 0.1.4 (user)"
  ]
}
```

**Comment by epavlic-revenue** [comment]
If `project-scratch-def.json` contains `"namespace": "SomeNamespace"`, `--no-namespace` is ignored and the scratch org will have a namespace.

**Comment by jshackell-sfdc** [comment]
This issue is fixed in version 2.41.8 -- enjoy!
